src/forms/ToggleSwitch.jsx

```diff
@@ -0,0 +1,119 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import cx from 'classnames';
+
+import { C_TEXT_PRIMARY, C_TEXT_SECONDARY, C_TEXT_HINT } from 'swarm-constants/dist/js/colorConstants.js';
+
+import Button from './Button';
+import Icon from '../media/Icon';
+import withToggleControl from '../utils/WithToggleControl';
+
+export const TOGGLE_SWITCH_CLASS = 'toggleSwitch';
+export const TOGGLE_SWITCH_ACTIVE_CLASS = `${TOGGLE_SWITCH_CLASS}--active`;
+export const TOGGLE_SWITCH_DISABLED_CLASS = `${TOGGLE_SWITCH_CLASS}--disabled`;
+
+export const KNOB_CLASS = `${TOGGLE_SWITCH_CLASS}-knob`;
+export const KNOB_ACTIVE_CLASS = `${KNOB_CLASS}--active`;
+export const KNOB_DISABLED_CLASS = `${KNOB_CLASS}--disabled`;
+
+export const LABEL_CLASS = `${TOGGLE_SWITCH_CLASS}-label`;
+
+/**
+ * @module ToggleSwitch
+ */
+class ToggleSwitch extends React.Component {
+	constructor (props) {
+		super(props);
+
+		this.toggle = this.toggle.bind(this);
+	}
+
+	toggle() {
+		!this.props.disabled && this.props.toggleActive();
+	}
+
+	render() {
+		const {
+			children,
+			className,
+			disabled,
+			isActive,
+			label,
+			labelClassName,
+			labelledBy,
+			name,
+			toggleActive, // eslint-disable-line no-unused-vars
+			...other
+		} = this.props;
```

Comment:
Seems like this could use additional props, like `onCallback` and `offCallback` that are called in `componentWillReceiveNewProps` (maybe?)  when the state of the switch changes. Thinking about the need to expand/collapse an `<AccordionPanel />` as the state of the `<ToggleSwitch />` is changed.

Reply:
If you think that's a correct observation, I don't think it matters if that change happens in this PR or in the `Accordion` update PR.

Reply:
Good idea, I'll add something to support this